axlearn/common/attention.py

```diff
@@ -3956,15 +3956,19 @@ def policy(prim, *_, **params):
     return policy
 
 
-SELF_ATTENTION_SAVE_PATTERN = ".*([qkvo]_proj|context)"
-FEED_FORWARD_SAVE_PATTERN = ".*linear[12]_.*"
+# Regex patterns for matching remat names
+class RematRegexSavePatterns(enum.Enum):
+    QKV_PROJ = r".*\.?(k|q|v)_proj"
+    LINEAR1_X = r".*\.?linear1_[01]"
+    SELF_ATTENTION = ".*([qkvo]_proj|context)"
+    FEED_FORWARD = ".*linear[12]_.*"
```

Comment:
Hmm, can we make the regex more composable, e.g., seems like `QKV_PROJ` is a subset of `SELF_ATTENTION`, and `LINEAR1_X` is a subset of `FEED_FORWARD`. Otherwise, we may get into an exponential number of save patterns. 

You can try to build something similar to the following. 
```suggestion
class RematRegexSavePatterns(enum.Enum):
    QKV_PROJ = r".*\.?(k|q|v)_proj"
    O_PROJ=r".*o_proj"
    CONTEXT=".*context"
    SELF_ATTENTION = re.compile( '|'.join( [O_PROJ,QKV_PROJ, CONTEXT]) )
```

Reply:
Added this, thanks